js/CMI5.js

```diff
@@ -0,0 +1,518 @@
+import Adapt from 'core/js/adapt';
+import logging from 'core/js/logging';
+
+class CMI5 {
+  constructor(xapi) {
+    this.xapi = xapi;
+  }
+
+  /**
+   * Configures the CMI5 launch by setting the launch parameters in the model,
+   * Setting the authentication token, actor, registration and activityId, fetching the CMI5 launch data.
+   * Fetches the agent profile data as per the cmi5 requirement.
+   * @returns {Promise<void>} A promise that resolves when the configuration is complete.
+   */
+  async configureCmi5Launch() {
+    // Set the launch parameters in the model
+    this.getLaunchParameters();
+    await this.setAuthToken();
+
+    // check to see if configuration has been passed in URL
+    this.xapi.xapiWrapper = window.xapiWrapper || window.ADL.XAPIWrapper;
+    this.xapi.xapiWrapper.changeConfig({
+      auth: `Basic ${this.xapi.get('_auth')}`,
+      activity_id: this.xapi.get('_activityId'),
+      strictCallbacks: true,
+    });
+    this.xapi.set({
+      actor: this.xapi.get('_actor'),
+      registration: this.xapi.get('_registration'),
+      activityId: this.xapi.get('_activityId'),
+    });
+    const cmi5LaunchData = await this.getCmi5LaunchData();
+    if (this.isMasteryScoreSet(cmi5LaunchData)) {
+      this.respectLMSMasteryScore(cmi5LaunchData);
+    }
+    if (cmi5LaunchData?.returnURL) {
+      Adapt.on('adapt:userExit', () =>
+        this.exitCourse(cmi5LaunchData.returnURL)
+      );
+    }
+
+    // https://github.com/AICC/CMI-5_Spec_Current/blob/quartz/cmi5_spec.md#110-xapi-agent-profile-data-model
+    const agentProfile = await this.getAgentProfile();
+    // Not currently using the agent profile data in the course
+    // but it's a cmi5 requirement to fetch it before sending statements
+    logging.info('Agent Profile: ', agentProfile);
+  }
+
+  /**
+   * Retrieves launch parameters from the URL and sets them as properties of the XAPI object.
+   */
+  getLaunchParameters() {
+    const params = new URLSearchParams(new URL(window.location).search);
+    if (params.size != 0) {
+      this.xapi.set({
+        _endpoint: params.get('endpoint'),
+        _fetch: params.get('fetch'),
+        _activityId: params.get('activityId'),
+        _actor: JSON.parse(decodeURIComponent(params.get('actor'))),
+        _registration: params.get('registration'),
+      });
+    }
```

Comment:
```suggestion
    if (params.size === 0) return;
    this.xapi.set({
      _endpoint: params.get('endpoint'),
      _fetch: params.get('fetch'),
      _activityId: params.get('activityId'),
      _actor: JSON.parse(decodeURIComponent(params.get('actor'))),
      _registration: params.get('registration'),
    });
```

Return early. Invert condition and return. Reduces nesting. Main use-case is at the bottom of the function. Please implement throughout.

We only use `===` not `==`. Please also correct throughout.